config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -48,6 +49,65 @@ function create_libvirt_image() {
 
 }
 
+# Function to create Libvirt pool and configure libvirt volume to upload qcow2 image
+function create_pool_and_volume() {
+
+    KVM_HOST_ADDRESS=$(echo "${LIBVIRT_URI}" | sed -E 's|^.*//([^/]+).*$|\1|')
+
+    if [[ -z "$KVM_HOST_ADDRESS" ]]; then
+        echo "Failed to extract IP address from the URI."
+        exit 1
+    fi
+
+    ssh "$KVM_HOST_ADDRESS" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i "/root/.ssh/id_rsa" "mkdir -p ${LIBVIRT_DIR_NAME}"
+    if [[ $? -eq 0 ]]; then
+        echo "Directory '$LIBVIRT_DIR_NAME' created successfully."
+    else
+        echo "Failed to create directory '$LIBVIRT_DIR_NAME' on '$KVM_HOST_ADDRESS'."
+        exit 1
+    fi
+ 
+    virsh -d 0 -c "${LIBVIRT_URI}" pool-info "${LIBVIRT_POOL}" > /dev/null 2>&1
+    if [ $? -eq 0 ]; then
+        echo "Pool '${LIBVIRT_POOL}' already exists."
+    else
+        virsh -d 0 -c "${LIBVIRT_URI}" pool-define-as "${LIBVIRT_POOL}" --type dir --target "${LIBVIRT_DIR_NAME}" \
+            || error_exit "Failed to define libvirt pool '${LIBVIRT_POOL}'."
+            
+        virsh -d 0 -c "${LIBVIRT_URI}" pool-start "${LIBVIRT_POOL}" \
+            || error_exit "Failed to start libvirt pool '${LIBVIRT_POOL}'."
+    fi
+    
+    virsh -d 0 -c "${LIBVIRT_URI}" vol-create-as --pool "${LIBVIRT_POOL}" \
+        --name "${LIBVIRT_VOL_NAME}" \
+        --capacity 20G \
+        --allocation 2G \
+        --prealloc-metadata \
+        --format qcow2 \
+        || error_exit "Failed to create libvirt volume '${LIBVIRT_VOL_NAME}' in pool '${LIBVIRT_POOL}'."
+
+    echo "Created libvirt pool and volume successfully."
+
+}
+
+# Function that check if defined volume and pool are already present
+function check_pool_and_volume_existence() {
+
+    echo "Checking existence of libvirt pool '${LIBVIRT_POOL}' and volume '${LIBVIRT_VOL_NAME}'..."
+    
+        virsh -d 0 -c "${LIBVIRT_URI}" pool-info "${LIBVIRT_POOL}" >/dev/null 2>&1
+        POOL_EXISTS=$?
+        virsh -d 0 -c "${LIBVIRT_URI}" vol-info --pool "${LIBVIRT_POOL}" "${LIBVIRT_VOL_NAME}" >/dev/null 2>&1
+        VOL_EXISTS=$?
+
+        if [ "$POOL_EXISTS" -eq 0 ] && [ "$VOL_EXISTS" -eq 0 ]; then
+            echo "Disclaimer: A Libvirt pool named '${LIBVIRT_POOL}' with volume '${LIBVIRT_VOL_NAME}' already exists on the KVM host. Image will be uploaded to same volume."
+        else
+            echo "Libvirt pool '${LIBVIRT_POOL}' or volume '${LIBVIRT_VOL_NAME}' does not exist. Proceeding to create..."
+            create_pool_and_volume
```

Comment:
A general suggestion. You can avoid mixing two functionalities in the same function. Since this function is specifically to check for pool and vol existence, return true or false.
Then based on return value call create_pool_and_volume.
Keeps the functions clean and scoped